File: HardwareInfoCollector/models/case.py
from models.hardware import *


class Case(Hardware):
    MotherBoardFormat: str
    MaxPowerSupplyLength: int
    MaxGpuLength: int
    MaxAirHeight: int
    MaxAioFansCount: int

    def __init__(self, product_type: int, model: str, price: int, link: str, motherboard_format: str,
                 max_power_supply_length: int, max_gpu_length: int, max_air_height: int, max_aio_fans_count: int):
        super().__init__(product_type, model, price, link)
        self.MotherBoardFormat = motherboard_format
        self.MaxPowerSupplyLength = max_power_supply_length
        self.MaxGpuLength = max_gpu_length
        self.MaxAirHeight = max_air_height
        self.MaxAioFansCount = max_aio_fans_count

    def __str__(self):
        return f'{self.ProductType} {self.Model} {self.Price} {self.Link} {self.MotherBoardFormat} ' \
               f'{self.MaxPowerSupplyLength} {self.MaxGpuLength} {self.MaxAirHeight} {self.MaxAioFansCount}'

File: BuildGenerator/src/Bll/Models/Hardware/Case.cs
namespace Bll.Models.Hardware;

public class Case : Abstractions.Hardware
{
    public double Price { get; set; }
    public string Link { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;

    public string MotherBoardFormat { get; set; }  = string.Empty;
    public int MaxPowerSupplyLength { get; set; }
    public int MaxGpuLength { get; set; }
    public int MaxAirHeight { get; set; }
    public int MaxAioFansCount { get; set; }

    public Case()
    {
        
    }
    
    public Case(double price, string link, string model, string motherboardFormat, int maxPowerSupplyLength,
        int maxGpuLength, int maxAirHeight, int maxAioFansCount)
    {
        Price = price;
        Link = link;
        Model = model;
        MotherBoardFormat = motherboardFormat;
        MaxPowerSupplyLength = maxPowerSupplyLength;
        MaxGpuLength = maxGpuLength;
        MaxAirHeight = maxAirHeight;
        MaxAioFansCount = maxAioFansCount;
    }
}